SwiftLeeds/Views/Components/FancyHeaderView.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  HeaderView.swift
+//  SwiftLeeds
+//
+//  Created by Kannan Prasad on 05/07/2022.
+//
+
+import SwiftUI
+
+struct FancyHeaderView: View {
+    let title: String
+    let foregroundImageURL: URL
+    let backgroundImageURL: URL
+    
+    private let foregroundImageWidth = 160.0
+    private let aspectRatio = 1.66
+    
+    @State var foregroudGroupViewHeight: CGFloat = .zero
+    
+    var body: some View {
+        Rectangle()
+            .foregroundColor(.clear)
+            .edgesIgnoringSafeArea(.top)
+            .aspectRatio(aspectRatio, contentMode: .fill)
+            .background(backgroundImage
+                            .aspectRatio(contentMode: .fill))
+            .overlay(foregroudGroupView,alignment: .center)
+            .padding(.bottom,foregroudGroupViewHeight/2)
+    }
+    
+    private var backgroundImage: some View {
+        AsyncImage(url: backgroundImageURL) { phase in
+            
+            switch phase {
+            case .empty:
+                ProgressView()
+            case .success(let image):
+                image
+                    .resizable()
+                    .aspectRatio(aspectRatio, contentMode: .fill)
+            case .failure(_):
+                ProgressView()
+                    .tint(.white)
+                    .opacity(0.5)
+                //TODO: Remove the loaded and set a default image
+            @unknown default:
+                ProgressView()
+                    .tint(.white)
+                    .opacity(0.5)
+            }
+        }
+    }
+    
+    private var foregroudGroupView: some View {
+        GeometryReader { geometry in
+            VStack{
```

Comment:
With a stack you shouldn't add padding this way to each element, you should instead take advantage of the built in spacing on a `VStack`

```suggestion
            VStack(spacing: Padding.stackGap) {
```

Reply:
Done